src/Microsoft.TestPlatform.Client/TestPlatform.cs

```diff
@@ -129,6 +133,13 @@ public ITestRunRequest CreateTestRunRequest(TestRunCriteria testRunCriteria, Pro
             var testHostManager = this.testHostProviderManager.GetTestHostManagerByRunConfiguration(testRunCriteria.TestRunSettings);
             testHostManager.Initialize(TestSessionMessageLogger.Instance, testRunCriteria.TestRunSettings);
 
+            // Allow TestRuntimeProvider to update source map, this is required for remote scenarios.
+            // If we run for specific tests, then we expect the test case object to contain correct source path for remote scenario as well
```

Comment:
I see no scenario where target device should change b/w discovery, & Execution. That being said, we currently discover UWP test on local machine, treating them via full CLR test, but this will give an opportunity to change it, once we can figure out UWP test via deploying container as hidden